src/modules/transaction/mod.rs

```diff
@@ -0,0 +1,14 @@
+use paperclip::actix::web;
+
+// mod data_provider;
+mod resources;
+mod schemas;
+
+pub(crate) fn register_services(app: &mut web::ServiceConfig) {
+    app.service(web::resource("/transaction").route(web::get().to(resources::get_transaction)))
+        .service(web::resource("/transactions").route(web::get().to(resources::get_transactions)))
+        .service(web::resource("/receipts").route(web::get().to(resources::get_receipts)))
+        .service(
+            web::resource("/receipts/actions").route(web::get().to(resources::get_action_receipts)),
+        );
```

Comment:
As I remember, you said you decided to change everything to path parameters. I don't see it here

Reply:
I changed the type and function definitions from query to param parameters, but forgot to add the param routes to this file. But upon doing more research into API designs, I believe it is more appropriate to use query parameters rather than path parameters to specify the endpoints. 


With the path params approach, we will increase the number of endpoints that a consumer of the endpoint need to familiarize themselves with e.g We need to be able to pull a transaction using a `transaction_hash`, as well as a `receipt_id`, and this will be implemented with two endpoints. One would be `transaction/:transaction_hash` and another for `transaction/:receipt_id` when in fact both of these endpoints pull the same resource. 

We want to make it as easy as possible for a user to know how to pull a specific resource, e.g. a transaction, transactions, receipts. 

In the query param design, it is easier for a user to see in **one place** that they can retrieve the transaction resource by either passing in a `transaction_hash` or `receipt_id` whereas a user of the API that supports the path params approach might know that a transaction can be retrieved using a `transaction_hash`, but without checking out the other endpoints, they would have no clue that they could also retrieve it via a `receipt_id`. 

Also, I think in the future we will need to incorporate query parameters to include filters like date_range to pull transactions that happened in a particular time range. So we should not restrict ourselves to using path parameters here for everything.

Also, Both [Etherscan](https://docs.etherscan.io/api-endpoints/accounts#get-internal-transactions-by-transaction-hash) and Figment support query parameters which make their API interface simpler.

Reply:
@khorolets, I am still open to discussing and changing the implementation to support path parameters if the team has strong opinions about that approach. @telezhnaya, @pkudinov what do you think? 

On one hand, it is better to remain consistent with the rest of the endpoint designs and on another, it is about making it easier to use the endpoint.

Reply:
> @khorolets, I am still open to discussing and changing the implementation to support path parameters if the team has strong opinions about that approach. @telezhnaya, @pkudinov what do you think?
> 
> On one hand, it is better to remain consistent with the rest of the endpoint designs and on another, it is about making it easier to use the endpoint.

@khorolets Could you please elaborate on what you mean by "why use RESTful if it looks like RPC?"?

Reply:
Confirming that we agreed to proceed with RPC-style paths from the beginning (RPC here as in "Remote Procedure Call Style" in general, not NEAR RPC specifically).

Reply:
> Confirming that we agreed to proceed with RPC-style paths from the beginning

@pkudinov Hmm, was the existing API style taken into account?

Just to fill it with some context here:

nearcore follows [JSON-RPC](https://www.jsonrpc.org/specification) (single path endpoint with JSON body), and it was the main pain of our Infra team trying to build metrics and proper throttling and scaling based on the queries as most of the tooling does not pick into the body params which is required when the type of request (e.g. block request or view-function call) is buried in the body of the HTTP request encoded as JSON. Also, there is no way to have nice interactive documentation or auto-generated client libraries for JSON-RPC.

Given the above problems, we decided to use RESTful design and use OpenAPI which has a lot of tooling (including interactive documentation and the opportunity to [generate client libraries in a myriad of programming languages](https://github.com/OpenAPITools/openapi-generator#62---openapi-generator-technical-committee)).

I feel the main problem here is miscommunication. Designing `/receipts/actions?receipt_id=xxx` path vs `/receipts/<receipt_id>/actions` is equal in time and effort. Lean and predictable API reduces the learning curve and helps developers to build production-quality apps fast.

Mixing RPC and RESTful styles will eventually break OpenAPI spec and users' expectations navigating the API. This will result in taking the worst parts of both worlds. Please, reconsider this decision.

Reply:
Thank you for the background on JSON-RPC @frol and for your insights @khorolets! It definitely makes more sense to stay consistent, so I changed the design back to being path params.